cluster/saltbase/salt/docker/init.sls

```diff
@@ -114,6 +114,36 @@ lxc-docker-{{ override_docker_ver }}:
       - lxc-docker-{{ override_docker_ver }}: /var/cache/docker-install/{{ override_deb }}
 {% endif %}
 
+# Default docker systemd unit file doesn't use an EnvironmentFile; replace it with one that does.
+{% if pillar.get('is_systemd') %}
+
+{{ pillar.get('systemd_system_path') }}/docker.service:
+  file.managed:
+    - source: salt://docker/docker.service
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - defaults:
+        environment_file: {{ environment_file }}
+
+# The docker service.running block below doesn't work reliably
```

Comment:
It doesn't work for you here because you do not have the docker.service file as a file to watch in the docker.service.running block.  We had a similar problem for the Kubelet in this area that I have a fix for here:
https://github.com/GoogleCloudPlatform/kubernetes/pull/10492

I think you can get rid of fix-service-docker and just add the watch on the file or a require on the file when running in a systemd setup.